add blog post about custom sd implementation

cstyan opened the pull request:
moving from the google doc to here, I've made a few more grammatical changes plus added more links and code blocks

cc @brian-brazil

brian-brazil commented on content/blog/2018-06-27-implementing-custom-sd.md:
```diff
@@ -0,0 +1,185 @@
+option isn’t feasible. In many cases the current SD implementations have been contributed by people
+outside the team and then not maintained or tested well. We want to commit to only providing direct
+integration with service discovery mechanisms that we know we can maintain, and that work as intended.
+For this reason, there is currently a moratorium on new SD integrations
```
full stop

brian-brazil commented on content/blog/2018-06-27-implementing-custom-sd.md:
```diff
@@ -0,0 +1,185 @@
+For this reason, there is currently a moratorium on new SD integrations
+
+However, we know there is still a desire to be able to integrate with other SD mechanisms, such as 
+Docker Swarm. Recently a small code change plus an example was committed to the documentation directory
```
link to the directory?

brian-brazil commented on content/blog/2018-06-27-implementing-custom-sd.md:
```diff
@@ -0,0 +1,185 @@
+However, we know there is still a desire to be able to integrate with other SD mechanisms, such as 
+Docker Swarm. Recently a small code change plus an example was committed to the documentation directory
+within the Prometheus repository for implementing a custom service discovery integration without having
+to merge it into the main Prometheus binary. The code change allows us to make use of the internal
```
We should probably make it clear here that we're not going to merge them into the docs directory either, we've had that confusion with the remote adapter example.

We should link to https://prometheus.io/docs/operating/integrations/#file-service-discovery too

cstyan commented:
addressed your comments, wasn't entirely sure what we wanted to say regarding the file_sd link but I am assuming we would potentially list known stable adapter/file_sd implementations on that page

brian-brazil commented on content/blog/2018-06-27-implementing-custom-sd.md:
```diff
@@ -0,0 +1,190 @@
+use of the adapter into the Prometheus 
+[discovery](https://github.com/prometheus/prometheus/tree/master/discovery) package. 
+
+Popular integrations that are implemented with the adapter code may end up listed 
```
Popularity isn't a requirement, I'd say that we list file_sd integrations such as those using this adapter on our integrations page.

cstyan commented:
@brian-brazil PTAL, I've rephrased the sentences you pointed out.

brian-brazil commented:
This looks good. Can you bump the date, as it's a bit too different by now?

cstyan commented:
:+1: updated the date to July 5 :)

brian-brazil commented:
Thanks!
